packages/design-system-docs/src/pages/components/Table/_Table.docs.scss

```diff
@@ -76,23 +76,13 @@ Style guide: components.table.react-tablerow
 */
 
 /*
-`<TableHeaderCell>`
+`<TableCell>`
 
-`TableHeaderCell` renders a `<th>` element.
+`TableCell` renders a `<th>` element when the parent component is a header or otherwise a `<td>` element.
```

Comment:
```suggestion
`TableCell` dynamically renders a `<th>` or `<td>` element based on the parent component or user specified `component` prop. By default `TableCell` will automatically render a `<th>` element if the parent component is `TableHead`, otherwise it will render a `<td>` element.
```